lib/src/utils/kratos_message.dart

```diff
@@ -1,178 +1,1398 @@
-  errorValidationMinimum(4000018),
-  errorValidationExclusiveMinimum(4000019),
-  errorValidationMaximum(4000020),
-  errorValidationExclusiveMaximum(4000021),
-  errorValidationMultipleOf(4000022),
-  errorValidationMaxItems(4000023),
-  errorValidationMinItems(4000024),
-  errorValidationUniqueItems(4000025),
-  errorValidationWrongType(4000026),
-  errorValidationDuplicateCredentialsOnOIDCLink(4000027),
-  errorValidationCredentialAlreadyUsedByAnotherAccount(4000028),
-  errorValidationMustBeEqualToConstant(4000029),
-  errorValidationConstFailed(4000030),
-  errorValidationPasswordTooSimilarToIdentifier(4000031),
-  errorValidationPasswordTooShort(4000032),
-  errorValidationPasswordTooLong(4000033),
-  errorValidationPasswordFoundInDataBreaches(4000034),
-  errorValidationNoAccountOrNoCodeSignInSetUp(4000035),
-  errorValidationTraitsDontMatchPreviouslyAssociated(4000036),
-
-  // Login Validation Error
-  errorValidationLogin(4010000),
-  errorValidationLoginFlowExpired(4010001),
-  errorValidationLoginNoStrategyFound(4010002),
-  errorValidationRegistrationNoStrategyFound(4010003),
-  errorValidationSettingsNoStrategyFound(4010004),
-  errorValidationRecoveryNoStrategyFound(4010005),
-  errorValidationVerificationNoStrategyFound(4010006),
-  errorValidationLoginRequestAlreadyCompleted(4010007),
-  errorValidationLoginCodeInvalidOrAlreadyUsed(4010008),
-  errorValidationLinkedCredentialsDoNotMatch(4010009),
-
-  // Registration Validation Error
-  errorValidationRegistration(4040000),
-  errorValidationRegistrationFlowExpired(4040001),
-  errorValidationRegistrationRequestAlreadyCompleted(4040002),
-  errorValidationRegistrationCodeInvalidOrAlreadyUsed(4040003),
-
-  // Settings Validation Error
-  errorValidationSettings(4050000),
-  errorValidationSettingsFlowExpired(4050001),
-
-  // Recovery Validation Error
-  errorValidationRecovery(4060000),
-  errorValidationRecoveryRetrySuccess(4060001),
-  errorValidationRecoveryStateFailure(4060002),
-  errorValidationRecoveryMissingRecoveryToken(4060003),
-  errorValidationRecoveryTokenInvalidOrAlreadyUsed(4060004),
-  errorValidationRecoveryFlowExpired(4060005),
-  errorValidationRecoveryCodeInvalidOrAlreadyUsed(4060006),
-
-  // Verification Validation Error
-  errorValidationVerification(4070000),
-  errorValidationVerificationTokenInvalidOrAlreadyUsed(4070001),
-  errorValidationVerificationRetrySuccess(4070002),
-  errorValidationVerificationStateFailure(4070003),
-  errorValidationVerificationMissingVerificationToken(4070004),
-  errorValidationVerificationFlowExpired(4070005),
-  errorValidationVerificationCodeInvalidOrAlreadyUsed(4070006),
-
-  // System Error
-  errorSystem(5000000),
-  errorSystemGeneric(5000001);
-
-  const KratosMessage(this.id);
-
-  final int id;
-
-  static KratosMessage forId(int id) {
-    return KratosMessage.values.firstWhereOrNull((error) => error.id == id) ??
-        KratosMessage.errorSystemGeneric;
+
+typedef ContextParameters = Map<String, dynamic>?;
```

Comment:
I don't think that's what Robert meant – let's keep this typedef nonnull, and assert that the map exists *before* passing it to the message